### PR TITLE
🎉 Source Amplitude: enable event stream time interval selection

### PR DESCRIPTION
**Pull request by mariamthiam**
## What
Enable the possibility to select events stream time interval  to resolve issue #21010

## How
Add new source parameter `event_time_interval` to make time interval selectionable for event stream

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Comment by misteryeo** [comment]
**Issue was linked to Harvestr Discovery:**  [Source Amplitude: Enable events stream `time_interval` selection  ](https://app.harvestr.io/components/0/list/JzHXh5CDx)

File: airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py [hunk 1]
```diff
@@ -148,7 +151,12 @@ class Events(IncrementalAmplitudeStream):
     compare_date_template = "%Y-%m-%d %H:%M:%S.%f"
     primary_key = "uuid"
     state_checkpoint_interval = 1000
-    time_interval = {"days": 1}
+
+    @property
+    def time_interval(self) -> dict:
+        return {
+            self.event_time_interval.get('size_unit'): self.event_time_interval.get("size")
+        }
```
Comment (evantahler) [inline airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py:159]: For users which are upgrading to this version of the connector from a pervious version, we probably should provide a default value if there wasn't `config.event_time_interval` set before

**Review by evantahler — CHANGES_REQUESTED** [review]
Thanks for the contribution!

Some things that will need to be added:
* A [changelog](https://github.com/airbytehq/airbyte/blob/master/docs/integrations/sources/amplitude.md) entry
* [Bump the version](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-amplitude/Dockerfile#L15) of the connector

This is probably a minor semver change, as this new config option is optional

File: airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py (hunk 1 above)
Comment (mariamthiam) [reply airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py:159]: There is a default value for `size_unit` and `size` in the spec so which means default value for `event_time_interval` also.

**Comment by sajarin** [comment]
hey @mariamthiam thanks for the contribution? Can you address the comments above by adding a change log entry and bumping the version in the dockerfile of the connector? Let me know if you need help with this or if you have any questions!

File: airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py (hunk 1 above)
Comment (mariamthiam) [reply airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py:159]: @evantahler can you update your review here !?

Comment (evantahler) [reply airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py:159]: Thank you for the explanation @mariamthiam! This PR still needs a version bump and a changelog entry, per the note above

**Comment by mariamthiam** [comment]
@evantahler  it done ✅

File: airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py (hunk 1 above)
Comment (evantahler) [reply airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py:177]: I think adding the default value in the config are helpful, but for users who are upgrading from the previous version of the connector, the config they will have from their previous version of the connector won't have any value for size_unit or size.  you can try this locally by removing the new object from your config file and trying a sync.    

I suggest you add some default behavior in the code here that matches the defaults in the connector spec

**Review by evantahler — CHANGES_REQUESTED** [review]

**Comment by evantahler** [comment]
The `/test` I'm running above did not include updating our config to demonstrate the connector upgrade case.  The config in use looks like:

```json
{
    "api_key": "xxx",
    "secret_key": "yyy",
    "start_date": "2023-02-01T00:00:00Z",
    "data_region": "Standard Server"
  }
```

File: airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py (hunk 1 above)
Comment (mariamthiam) [reply airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py:177]: I have added it 👍

File: airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py [hunk 2]
```diff
@@ -24,8 +24,11 @@
 class AmplitudeStream(HttpStream, ABC):
     api_version = 2
 
-    def __init__(self, data_region: str, **kwargs):
+    def __init__(self, data_region: str, event_time_interval: dict = None, **kwargs):
+        if event_time_interval is None:
+            event_time_interval = {"size_unit": "days", "size": 1}
```
Comment (mariamthiam) [inline airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py:29]: @evantahler default value is set here

**Comment by mariamthiam** [comment]
Sorry this should be the last fix can you rerun the tests @evantahler ?

**Review by evantahler — APPROVED** [review]
🥳 I want to ask one person for a review, and then we can merge this in!

**Review by bazarnov — APPROVED** [review]
Looks good. LGTM!

**Comment by evantahler** [comment]
Thank you @mariamthiam! I'm getting this published and merged in now.

**Pull request merged**